os/drivers/lcd/mipi_lcd.c

```diff
@@ -0,0 +1,414 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/lcd/lcd.h>
+#include <tinyara/lcd/lcd_dev.h>
+#include <tinyara/lcd/mipi_lcd.h>
+#include <tinyara/mipidsi/mipi_dsi.h>
+#include <tinyara/mipidsi/mipi_display.h>
+#include <debug.h>
+#include <assert.h>
+
+#define REGFLAG_DELAY                       0xFC
+#define REGFLAG_END_OF_TABLE                0xFD	// END OF REGISTERS MARKER
+
+#if !defined(CONFIG_LCD_MAXPOWER) || CONFIG_LCD_MAXPOWER < 1
+#define CONFIG_LCD_MAXPOWER 100
+#endif
+
+struct mipi_lcd_dev_s {
+	/* Publicly visible device structure */
+
+	struct lcd_dev_s dev;
+
+	FAR struct mipi_dsi_device *dsi_dev;
+
+	u8 *lcd_img_buffer[MAX_NO_PLANES];
+	//u8 *BackupLcdImgBuffer;
+	int fb_alloc_count;
+	uint8_t power;
+	struct mipi_lcd_config_s *config;
+};
+
+static struct mipi_lcd_dev_s g_lcdcdev;
+
+typedef struct lcm_setting_table {
+	u8 cmd;
+	u16 count;
+#if defined(CONFIG_LCD_ST7785)
+	u8 para_list[28];
+#elif defined(CONFIG_LCD_ST7701)
+	u8 para_list[16];
+#endif
+} lcm_setting_table_t;
+
+static const lcm_setting_table_t lcd_init_cmd_g[] = {	/* DCS Write Long */
+#if defined(CONFIG_LCD_ST7701)
```

Comment:
Let's make a new header file for this.
```
in st7701_init_cmd.h

static const lcm_setting_table_t lcd_init_cmd_g[] = { /* DCS Write Long */
	/* ST7701S Reset Sequence */
	/* LCD_Nreset (1); Delayms (1); */
	/* LCD_Nreset (0); Delayms (1); */
	/* LCD_Nreset (1); Delayms (120); */
	{0x11, 0, {0x00}},
	{REGFLAG_DELAY, 120, {}},	/* Delayms (120) */
        ....
};

in mipi_lcd.c

#ifdef CONFIG_LCD_ST7701
#include "st7701_init_cmd.h"
#elif defined(CONFIG_LCD_ST7789)
....
#endif
```
Let's split common code and chipset specific value. 
Same as the size of the `para_list`. You can define common name for the size (define it in each header and use the definition here)